### ISSUE TITLE
Create OpenAPI rule for enums that don't match the attribute type

### ISSUE THREAD

**Issue by rossmcdonald** [stoplightio/spectral#570]
As an example, if I have a numeric attribute in a JSON schema object. I want to ensure that any enumerations of that attribute are of the same numeric type (ie, not a string). It should be flagged as an error when this occurs.

For example, the JSON schema below should trigger an error:

```yaml
  a_model:
    type: object
    properties:
      id:
        type: integer
        description: Unique asset identifier
        enum:
          - 'a string!'
        format: int64
```

Since the `id` field is of type integer but the enum is a string. While this:

```yaml
  a_model:
    type: object
    properties:
      id:
        type: integer
        description: Unique asset identifier
        enum:
          - 123
        format: int64
```

Should not.

Created from https://github.com/stoplightio/studio/issues/116

**Comment by nulltoken** [comment]
@rossmcdonald Could you please add a minimal repro case that should be detected by the new rule to be created?

**Comment by rossmcdonald** [comment]
@nulltoken Sorry about that. Description updated with a minimal repro case. Let me know if anything is unclear.

**Comment by nulltoken** [comment]
@rossmcdonald :+1: Super duper clear!

**Comment by nulltoken** [comment]
@rossmcdonald Could this be reworded as follows: "When a member is described through both an enum and a format, and that format is known to be validatable (https://github.com/epoberezkin/ajv#formats along with https://github.com/stoplightio/spectral/blob/develop/src/functions/schema.ts#L45-L49), then all members of the enum should respect that format"?

**Comment by rossmcdonald** [comment]
@nulltoken Yep, that's exactly it.

**Comment by nulltoken** [comment]
@philsturgeon started to work on this

**Comment by nulltoken** [comment]
@rossmcdonald #913 should (hopefully) fit your request.
